subprojects/platform-native/src/testFixtures/groovy/org/gradle/nativeplatform/fixtures/SingleToolChainTestRunner.java

```diff
@@ -21,48 +21,50 @@
         List<AvailableToolChains.ToolChainCandidate> toolChains = AvailableToolChains.getToolChains();
-        if (enableAllToolChains) {
-            for (AvailableToolChains.ToolChainCandidate toolChain : toolChains) {
-                if (!toolChain.isAvailable()) {
-                    throw new RuntimeException(String.format("Tool chain %s is not available.", toolChain.getDisplayName()));
-                }
-                add(new ToolChainExecution(toolChain, isRespectsInstalledConstraint(toolChain)));
+
+        for (AvailableToolChains.ToolChainCandidate toolChain : toolChains) {
+            if (!toolChain.isAvailable()) {
+                throw new RuntimeException(String.format("Tool chain %s is not available.", toolChain.getDisplayName()));
             }
-        } else {
-            boolean hasEnabled = false;
-            for (AvailableToolChains.ToolChainCandidate toolChain : toolChains) {
-                if (!hasEnabled && toolChain.isAvailable() && isRespectsInstalledConstraint(toolChain)) {
-                    add(new ToolChainExecution(toolChain, true));
-                    hasEnabled = true;
-                } else {
-                    add(new ToolChainExecution(toolChain, false));
-                }
+            if (canUseToolChain(toolChain)) {
+                add(new ToolChainExecution(toolChain));
             }
         }
     }
 
-    private boolean isRespectsInstalledConstraint(AvailableToolChains.ToolChainCandidate toolChain) {
-        RequiresInstalledToolChain toolChainRequirement = target.getAnnotation(RequiresInstalledToolChain.class);
-        return toolChain.meets(toolChainRequirement == null ? ToolChainRequirement.AVAILABLE : toolChainRequirement.value());
+    // TODO: This exists because we detect all available native tool chains on a system (clang, gcc, swiftc, msvc)
+    // Many of our old tests assume that available tool chains can compile many/most languages, so they do not try to
+    // restrict the required set of tool chains.
+    // The swiftc tool chain can build _only_ Swift, so tests must opt-in to see the swiftc tool chain.
+    // Our multi-test runner is smart enough to disable tests that do not meet our requirements, but since many
+    // of the old tests do not have requirements, we need to remove tool chains that do not meet the test classes
+    // requirements (if there are any) early.
+    // In the future... we want to either split apart the swiftc tool chain into a separate test runner/tool chain
+    // detector or go back to old tests and annotate them with tool chains requirements (like NOT_SWIFTC) or
+    // something more involved like declaring which features are needed for the test (I need a tool chain for
+    // C and a runtime like such-and-such).
+    private boolean canUseToolChain(AvailableToolChains.ToolChainCandidate toolChain) {
+        if (toolChain.meets(ToolChainRequirement.SWIFTC)) {
```

Comment:
We shouldn't leave this here very long. It would be better to have a coarse grained "c family" tool chain requirement and attach this to the tests that don't declare anything. As a first step, we can assume this here as the default, rather than baking in the opposite assumption here.